sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -500,8 +501,18 @@ final class UnboundedWindowFunctionFrame(
     if (processor != null) {
       processor.initialize(rows.length)
       val iterator = rows.generateIterator()
-      while (iterator.hasNext) {
-        processor.update(iterator.next())
+      if (jumpToEnd) {
+        var lastRow = EmptyRow
+        while (iterator.hasNext) {
+          lastRow = iterator.next()
```

Comment:
Is there any better way to get the last element in this case than iterate over all of them? maybe not given the unusual nature of the dat backing ExternalAppendOnlyUnsafeRowArray

Reply:
`ExternalAppendOnlyUnsafeRowArray` cannot get the last element directly.